AmethystAPI/src/amethyst/runtime/mod/Mod.cpp

```diff
@@ -0,0 +1,147 @@
+#include "Mod.h"
+#include <fstream>
+
+Mod::Mod(std::string modName)
+{
+    this->modName = modName;
+    fs::path dllPath = GetTempDll();
+
+    // Loads the mod in a temporary directory so that the original DLL can still be built to
+    hModule = LoadLibrary(dllPath.string().c_str());
+    if (hModule == NULL) {
+        DWORD error = GetLastError();
+
+        switch (error) {
+        case 0x5:
+            Log::Error("[AmethystRuntime] '{}' does not have the required privileges!", dllPath.string());
```

Comment:
If you wouldn't mind, could you change these to use the new assert macro. I know it's from before your PR

Reply:
I'll do it